deploy/AludelV1Template.ts

```diff
@@ -6,23 +6,29 @@ import { ALUDEL_V1_VANITY_ADDRESS } from "../constants";
 const deployFunc = async function ({
   ethers,
   deployments,
+  getNamedAccounts,
 }: HardhatRuntimeEnvironment) {
   const { get, log } = deployments;
+  const { deployer } = await getNamedAccounts();
 
   const deployedFactory = await get("AludelFactory");
-  const factory = await ethers.getContractAt(
-    deployedFactory.abi,
-    deployedFactory.address
-  );
+  const factory = (
+    await ethers.getContractAt(deployedFactory.abi, deployedFactory.address)
+  ).connect(await ethers.getSigner(deployer));
 
   log("Adding disabled AludelV1 empty template to factory");
   // this is only meant to add previous programs, therefore it's disabled from the start
   try {
     await (
       await factory.addTemplate(ALUDEL_V1_VANITY_ADDRESS, "AludelV1", true)
     ).wait();
-  } catch {
-    log("WARNING: AludelV1 was already added");
+  } catch (err) {
+    // cast sig 'TemplateAlreadyAdded()'
+    if (err.data === "0xf298693e") {
```

Comment:
I don't mind this but it's weird that typechain doesn't provide this kind of data out of the box.

Reply:
I tought about scraping it from the ABI to get an error in case the error is ever renamed, but that wouldn't be immune to bugs of its own. One more reason for testing in Solidity I guess 🙃